app/models/step_by_step_page.rb

```diff
@@ -30,7 +30,7 @@ def has_draft?
   end
 
   def scheduled_for_publishing?
-    has_draft? && scheduled_at.present?
+    has_draft? && scheduled_at.present? && status == "scheduled"
```

Comment:
What about using a [`StringInquirer`](https://api.rubyonrails.org/v5.2/classes/ActiveSupport/StringInquirer.html) to turn status into methods? such as `status.scheduled?`

Reply:
Updated and fixed-up